### PR TITLE
Assorted fixes and improvements

### PR DESCRIPTION
**Pull request by vladmoroz**
- Add a comprehensive Reset component
- Ensure all elements that have padding or border styles use `box-sizing: border-box`
- Fix buggy disabled cursor styles
- Tweak the background color of all `variant="soft"` menu items
- Fix a regression when values for `type: 'string'` prop defs wouldn't work (props like `gridColumn`, `gridRow`, etc)
- Fix an issue when Scroll Area would be unable to stretch to 100% height when informed by the parent’s auto height (originally battle-tested via https://github.com/workos/workos/pull/23554, no regressions from that)
- Change the default Tooltip delay duration to 200ms
- Map size 2 loading button to size 2 spinner
- Rename DataList.Data to DataList.Value

### CODE REVIEW

File: packages/radix-ui-themes/src/components/reset.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import { Slot } from '@radix-ui/react-slot';
+import { extractProps, marginPropDefs, requireReactElement } from '../helpers';
+
+import type { MarginProps, PropsWithoutRefOrColor } from '../helpers';
+
+type ResetElement = React.ElementRef<'div'>;
+interface ResetProps extends PropsWithoutRefOrColor<typeof Slot>, MarginProps {}
```
Comment (vladmoroz) [inline packages/radix-ui-themes/src/components/reset.tsx:9]: Would having MarginProps be a nice DX to help you build custom components on top of Reset?

File: packages/radix-ui-themes/src/components/base-menu.css [hunk 2]
```diff
@@ -30,17 +31,18 @@
...
-  cursor: var(--cursor-menu-item);
   scroll-margin: var(--base-menu-content-padding) 0;
 
   /* Fix sticky text highlighting after selection in Firefox */
   user-select: none;
 
+  /* Cursors */
+  cursor: var(--cursor-menu-item);
   &:where([data-disabled]) {
-    pointer-events: none;
```
Comment (vladmoroz) [inline packages/radix-ui-themes/src/components/base-menu.css:43]: - Why was there pointer-events: none?
- In theory, we are supposed to use disabled cursors here, but they don't feel right in menus

File: packages/radix-ui-themes/src/components/base-dialog.css [hunk 3]
```diff
@@ -18,7 +18,12 @@
 }
 
 .rt-BaseDialogScrollPadding {
-  margin: auto;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  width: 100%;
+  box-sizing: border-box;
```
Comment (vladmoroz) [inline packages/radix-ui-themes/src/components/base-dialog.css:26]: Previous styles weren't encouraging the dialog content to expand to its full width

File: packages/radix-ui-themes/src/components/base-button.css [hunk 4]
```diff
@@ -220,7 +219,6 @@
     }
   }
   &:where([data-disabled]) {
-    cursor: var(--cursor-disabled);
```
Comment (vladmoroz) [inline packages/radix-ui-themes/src/components/base-button.css:223]: Handled by the reset (just like the cursor in the normal state)

File: packages/radix-ui-themes/src/components/reset.css [hunk 5]
```diff
@@ -2,55 +2,163 @@
...
+  text-decoration: none;
+}
+.rt-reset:where(img) {
+  display: block;
+  max-width: 100%;
+}
+.rt-reset:where(svg) {
+  display: block;
+  max-width: 100%;
+  flex-shrink: 0;
```
Comment (vladmoroz) [inline packages/radix-ui-themes/src/components/reset.css:120]: Fun fact, `img` doesn't shrink in flex layouts, but `svg` does

File: packages/radix-ui-themes/src/components/base-dialog.css (hunk 3 above)
Comment (vladmoroz) [reply packages/radix-ui-themes/src/components/base-dialog.css:26]: Compare the dialogs
https://themes-playground-modulz.vercel.app/test-inset
https://themes-playground-git-vlad-a-bag-of-fixes-modulz.vercel.app/test-inset

File: packages/radix-ui-themes/src/components/index.ts [hunk 6]
```diff
@@ -149,7 +149,7 @@ export { Callout, CalloutRoot, CalloutIcon, CalloutText } from './callout';
 export * from './callout.props';
 export { Card } from './card';
 export * from './card.props';
-export { DataListRoot, DataListData, DataListItem, DataListLabel } from './data-list';
+export { DataListRoot, DataListValue, DataListItem, DataListLabel } from './data-list';
```
Comment (benoitgrelard) [inline packages/radix-ui-themes/src/components/index.ts:152]: Ultra nit:
```suggestion
export { DataListRoot, DataListItem, DataListLabel, DataListValue } from './data-list';
```

File: packages/radix-ui-themes/src/components/reset.tsx (hunk 1 above)
Comment (benoitgrelard) [reply packages/radix-ui-themes/src/components/reset.tsx:9]: That's nice yeah I like that.

**Review by benoitgrelard — APPROVED** [review]
Looks good, just one tiny nit.

Also, the branch needs rebasing, so to ease my review I restored the original target branch but it needs switching to `main` before we merge.

File: packages/radix-ui-themes/src/components/index.ts (hunk 6 above)
Comment (vladmoroz) [reply packages/radix-ui-themes/src/components/index.ts]: Something satisfying in the parts order matching the alphabetical one now 🙂

**Comment by benoitgrelard** [comment]
@vladmoroz this wasn't rebased/merged into the correct branch.

**Pull request merged**